snapd.go

```diff
@@ -267,6 +267,13 @@ func action(ctx *cli.Context) {
 	// same variables in that configuration
 	applyCmdLineFlags(cfg, ctx)
 
+	// Validate log level and trust level settings for snapd
+	validateLevelSettings(cfg.LogLevel, cfg.Control.PluginTrust)
+
+	// Switch log level to user defined
+	log.Info("setting log level to: ", l[cfg.LogLevel])
```

Comment:
This info will get logged no matter what the user sets as the log level. Probably need to move it to after we set the level and maybe change the phrasing to "set log level to: ".

Another issue is that if the logpath is set this would still be sent to stdout. If logpath is set we want all logging to go to the file specified. I think there is some re-ordering that would need to happen there also.